nbextensions/usability/toc2/main.js

```diff
@@ -274,77 +273,60 @@ var table_of_contents = function () {
     //proces_cell_toc();
     
     var cell_toc_text = "# Table of Contents\n <p>";
-    var depth = 1;
-    var li;
-    $("#notebook").find(":header").map(function (i, h) {
-      var level = parseInt(h.tagName.slice(1), 10);
+    var depth = 1; //var depth = ol_depth(ol);
+    var li= ul;//yes, initialize li with ul! 
+    var all_headers= $("#notebook").find(":header");
+    var min_lvl=1, lbl_ary= [];
+    for(; min_lvl <= 6; min_lvl++){ if(all_headers.is('h'+min_lvl)){break;} }
+    for(var i= min_lvl; i <= 6; i++){ lbl_ary[i - min_lvl]= 0; }
+
+    //loop over all headers
+    all_headers.each(function (i, h) {
+      var level = parseInt(h.tagName.slice(1), 10) - min_lvl + 1;
       // skip below threshold
-      if (level > threshold) return;
+      if (level > threshold){ return; }
       // skip headings with no ID to link to
-      if (!h.id) return;
+      if (!h.id){ return; }
```

Comment:
BTW: IMO it would be nice to a) add IDs so that you wouldn't need to add them on your own if you display H1 from a  kernel and b) to check that a ID wasn't already included in the list if if it was, do some magic that the links work anyways...

[The last is a problem when including something like this: `"analysis 1": "plots", "stats"; analysis 2: "plots", "stats"` -> the links for both "stats" all go to the first one because the notebook does not render unique IDs -> https://github.com/jupyter/notebook/issues/77]